planner/cascades/transformation_rules.go

```diff
@@ -111,6 +104,11 @@ func GetPattern(id TransformationID) *memo.Pattern {
 type PushSelDownTableScan struct {
 }
 
+// ID implements Transformation interface.
+func (r *PushSelDownTableScan) ID() TransformationID {
+	return rulePushSelDownTableScan
```

Comment:
can we generate this ID by some default function?

Reply:
I have no other ideas. Any suggestions? cc @eurekaka

Reply:
You can change the `ID()` method to something like `String()`, e.g,
```golang
func (r *PushSelDownTableScan) String() string {
    return "push_sel_down_table_scan"
}
```
then you can remove all the `TransformationID` things. The `patternMap` need to be changed to `map[string]Transformation` type as well.

Also, we'd better re-fill `patternMap` in `ResetTransformationRules`.

Reply:
Or we can use `map[Transformation]*memo.Pattern` as type of `patternMap`, then we don't need `String()` at all, but I prefer `String()` approach because it is more debug-friendly.

Reply:
The reason why I introduced `TransformationID` before is that using `[]*memo.Pattern` for `patternMap` is better than `map[string]*memo.Pattern`. cc @winoros

Reply:
We don't even need id if the pattern is stored inside the rule.

Reply:
How about:
```

Reply:
Or just create the rule by function `NewXXXX` and init everything there.